crates/energy/benches/nearby.rs

```diff
@@ -0,0 +1,142 @@
+use de_energy::{update_nearby_recv, EnergyReceiver, NearbyUnits};
+use de_index::{EntityIndex, LocalCollider};
+use de_objects::ObjectCollider;
+use parry3d::math::{Isometry, Vector};
+use parry3d::shape::{Cuboid, TriMesh};
+
+const MAP_SIZE: f32 = 2000.;
+const DISTANCE_FROM_MAP_EDGE: f32 = 100.;
+
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct UpdateUnits;
+
+#[derive(Component)]
+struct UnitNumber(u32);
+
+fn update_index(
+    mut index: ResMut<EntityIndex>,
+    moved: Query<(Entity, &Transform), Changed<Transform>>,
+) {
+    for (entity, transform) in moved.iter() {
+        let position = Isometry::new(
+            transform.translation.into(),
+            transform.rotation.to_scaled_axis().into(),
+        );
+        index.update(entity, position);
+    }
+}
+
+fn load_points(number: u32) -> Vec<Vec2> {
+    let mut points_path: PathBuf = env!("CARGO_MANIFEST_DIR").into();
+    points_path.push("test_data");
+    points_path.push(format!("{number}-points.txt"));
+    let reader = BufReader::new(File::open(points_path).unwrap());
+
+    let mut points = Vec::with_capacity(number as usize);
+    for line in reader.lines() {
+        let line = line.unwrap();
+        let mut numbers = line.split_whitespace();
+        let x: f32 = numbers.next().unwrap().parse().unwrap();
+        let y: f32 = numbers.next().unwrap().parse().unwrap();
+        points.push((MAP_SIZE - DISTANCE_FROM_MAP_EDGE) * Vec2::new(x, y));
+    }
+    points
+}
+
+fn init_world_with_entities_moving(world: &mut World, num_entities: u32) {
+    let mut index = EntityIndex::new();
+    let points = load_points(num_entities);
+
+    for (i, point) in points.into_iter().enumerate() {
+        let x: f32 = point.x;
+        let y: f32 = point.y;
+
+        let collider = LocalCollider::new(
+            ObjectCollider::from(TriMesh::from(Cuboid::new(Vector::new(3., 3., 4.)))),
+            Isometry::new(Vector::new(x, 0., y), Vector::identity()),
+        );
+
+        let entity = world
+            .spawn((
+                Transform::from_xyz(x, y, 0.0),
+                EnergyReceiver,
+                NearbyUnits::default(),
+                UnitNumber(i as u32),
+            ))
+            .id();
+
+        index.insert(entity, collider);
+    }
+
+    world.insert_resource(index);
+}
+
+/// Move entities in circles of radius N / 2.
+fn move_entities_in_circle(time: Res<Time>, mut query: Query<&mut Transform>) {
+    for mut transform in query.iter_mut() {
+        // Change direction (counter)clockwise based on entity_mum % 2 == 0
+        let direction = if transform.translation.x % 2. == 0. {
+            1.
+        } else {
+            -1.
+        };
+
+        let t = time.elapsed_seconds();
+        let radius = DISTANCE_FROM_MAP_EDGE / 2.;
+
+        let x = radius * (t * 0.5 * direction).sin();
+        let y = radius * (t * 0.5 * direction).cos();
+
+        transform.translation.x = x;
```

Comment:
This should be `center.x + x` (each entity needs to have a center) so that all entities doesn't circle around the same center added by `init_world_with_entities_moving`.